File: angular_workspace/pfiva-web/src/app/data-model/message/Message.ts
import { User } from "../User";


export class Message {
	private id: number;
	private messageText: string;
	private deliveryDateTime: string;
	private messageStatus: string;
	private users: User[];
	private topic: string;

	public get Id(): number {
		return this.id;
	}
	public set Id(value: number) {
		this.id = value;
	}
	public get MessageText(): string {
		return this.messageText;
	}
	public set MessageText(value: string) {
		this.messageText = value;
	}
	public get DeliveryDateTime(): string {
		return this.deliveryDateTime;
	}
	public set DeliveryDateTime(value: string) {
		this.deliveryDateTime = value;
	}
	public get MessageStatus(): string {
		return this.messageStatus;
	}
	public set MessageStatus(value: string) {
		this.messageStatus = value;
	}
	public get Users(): User[] {
		return this.users;
	}
	public set Users(value: User[]) {
		this.users = value;
	}
	public get Topic(): string {
		return this.topic;
	}
	public set Topic(value: string) {
		this.topic = value;
	}
}

File: workspace/data-model/src/main/java/de/pfiva/data/model/message/Message.java
package de.pfiva.data.model.message;

import java.io.Serializable;
import java.util.List;

import de.pfiva.data.model.User;

public class Message implements Serializable {

	private static final long serialVersionUID = 1L;
	
	private int id;
	private String messageText;
	private String deliveryDateTime;
	private MessageStatus messageStatus;
	private String topic;
	private List<User> users;
	
	public enum MessageStatus {
		DELIVERED, PENDING, CANCELLED, FAILED
	}
	
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getMessageText() {
		return messageText;
	}
	public void setMessageText(String messageText) {
		this.messageText = messageText;
	}
	public String getDeliveryDateTime() {
		return deliveryDateTime;
	}
	public void setDeliveryDateTime(String deliveryDateTime) {
		this.deliveryDateTime = deliveryDateTime;
	}
	public MessageStatus getMessageStatus() {
		return messageStatus;
	}
	public void setMessageStatus(MessageStatus messageStatus) {
		this.messageStatus = messageStatus;
	}
	public String getTopic() {
		return topic;
	}
	public void setTopic(String topic) {
		this.topic = topic;
	}
	public List<User> getUsers() {
		return users;
	}
	public void setUsers(List<User> users) {
		this.users = users;
	}
	@Override
	public String toString() {
		return "Message [id=" + id + ", messageText=" + messageText + ", deliveryDateTime=" + deliveryDateTime
				+ ", messageStatus=" + messageStatus + ", topic=" + topic + ", users=" + users + "]";
	}
}
